configuration/src/main/java/com/synopsys/integration/configuration/config/PropertyConfiguration.java

```diff
@@ -172,6 +177,20 @@ public boolean wasKeyProvided(@NotNull final String key) {
         return getRaw(keys::contains);
     }
 
+    @NotNull
+    public Map<String, String> getRawKeyValueMap(@NotNull final Set<Property> properties) {
```

Comment:
If you actually want a `Map<Property, String>` why not just return that here? It seems you do a lot of hoop jumping later to turn it into that. Which I think would be expected if you had to add a lot to support it but we are just using the existing getRaw, so I don't see why it couldn't just return what you want.

Reply:
I determined that we actually just want a map of key -> value, and don't need the Property.  The reason I thought we needed it was for PropertyConfigurationHelpContext to determine source, but I added that getPropertySource(key) method to PropertyConfiguration so all PCHC needs is the key.

Reply:
It seems surprising then you have the getCorrespondingProperty method then.

Reply:
getCorrespondingProperty is strictly used by Map<Property, String> maskRawValues, which should just be deleted.  So it would go as well.